spec/lib/dispatch-rider/logging/lifecycle_logger_spec.rb

```diff
@@ -1,72 +1,161 @@
 
-    it "calls logger with info" do
-      expect(logger).to receive(:info).with(string_to_log)
-    end
-  end
+    context 'when additional_info_interjector is customized' do
+      let(:additional_info_interjector) do
+        -> (message) do
+          message[:employee_id] = 47
+          message[:account_id] = 42
+          message
+        end
+      end
 
-  context "wrap_handling" do
-    context "block runs successfully" do
-      let(:block) { Proc.new { true } }
-      after { subject.wrap_handling(message, &block) }
+      before do
+        DispatchRider.config.additional_info_interjector = additional_info_interjector
+      end
 
-      it "logs start" do
-        expect(subject).to receive(:log_start).with(message)
+      describe 'log_error_handler_fail' do
+        let(:expected_info) do
+          {
+            phase: 'error_handler_fail',
+            guid: '123',
+            subject: 'test',
+            body: { some: 'key' },
+            exception: { class: 'StandardError', message: 'something failed' },
+            employee_id: 47,
+            account_id: 42,
+          }.deep_stringify_keys.to_json
+        end
+
+        it "calls logger with error" do
+          expect(logger).to receive(:error).with(expected_info)
+          subject.log_error_handler_fail fs_message, exception
+        end
       end
 
-      it "logs success" do
-        expect(subject).to receive(:log_success).with(message)
+      describe "log_got_stop" do
+        let(:expected_info) do
+          {
+            phase: 'stop',
+            guid: '123',
+            subject: 'test',
+            body: { some: 'key' },
+            reason: reason,
+            employee_id: 47,
+            account_id: 42,
+          }.deep_stringify_keys.to_json
+        end
+
+        it "calls logger with info" do
+          expect(logger).to receive(:info).with(expected_info)
+          subject.log_got_stop reason, fs_message
+        end
       end
 
-      it "logs complete" do
-        expect(subject).to receive(:log_complete).with(message, an_instance_of(Float))
+      describe "wrap_handling" do
+        let(:expected_start_info) do
+          {
+            phase: 'start',
+            guid: '123',
+            subject: 'test',
+            body: { some: 'key' },
+            employee_id: 47,
+            account_id: 42,
+          }.deep_stringify_keys.to_json
+        end
+
+        let(:expected_succeed_info) do
+          {
+            phase: 'success',
+            guid: '123',
+            subject: 'test',
+            body: { some: 'key' },
+            employee_id: 47,
+            account_id: 42,
+          }.deep_stringify_keys.to_json
+        end
+
+        let(:expected_complete_info) do
+          /{\"phase\"\:\"complete\",\"guid\"\:\"123\",\"subject\"\:\"test\",\"body\"\:{\"some\"\:\"key\"},\"duration\"\:\d+(?:.\d+)?,\"employee_id\"\:47,\"account_id\"\:42}/
```

Comment:
I'm consciously leaving this here unless someone knows how to do multiline, non breaking regex matchers. I only know how to do those for strings.